modules/expansionhunter/main.nf

```diff
@@ -0,0 +1,45 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process EXPANSIONHUNTER {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::expansionhunter=4.0.2" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/expansionhunter:4.0.2--he785bd8_0"
+    } else {
+        container "quay.io/biocontainers/expansionhunter:4.0.2--he785bd8_0"
+    }
+
+    input:
+    tuple val(meta), path(bam), path(bai)
+    path fasta
+    path variant_catalog
+
+    output:
+    tuple val(meta), path("*.vcf"), emit: vcf
+    path "*.version.txt"          , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def sex = (meta.sex == 'male' || meta.sex == 1 || meta.sex == 'XY') ? "male" : "female"
```

Comment:
I can see that you use meta.sex, we have meta.gender in sarek, should we try to have some cohesion at the nf-core level here?

Reply:
We mainly use gender for CNV, so we need either XX or XY, but I actually just use a string, so it can be anything.
We're doing things a bit differently in sarek, so the csv file is handled only in groovy (need to add more checks...).
And we have some optional fields in our header, but basically, if the gender is present, at the moment, we don't check how it's encoded, that's definitively a good idea to have, otherwise it's "NA".
https://github.com/nf-core/sarek/blob/6462bc81115b201792ed35e8252112204ed5c602/workflows/sarek.nf#L565

Reply:
expansionhunter is only used in human anyway, right?

Reply:
You need a catalog of know sites and those are, as far as I know, only available for human data right now. 

It might be tricky to have `--sex` in options.args since it's sample specific and you might want to run multiple samples in a together (like ina trio). Could options.args be set to different values for different samples?

Reply:
That's a valid point.
I'm happy with the current PR.
It's already a very good one.
We'll check for improvements later.
Maybe we'll figure a nice way to call the meta map in the options.args in the future.

Reply:
I'm not aware of any smart way to set options.args to different values for different samples, I'm afraid they're not doing that in the rnaseq pipeline, and we're doing that in sarek either.